benchmarking/src/run.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022-2023 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+//! Library for running and reporting benchmark results.
+
+use std::{collections::HashMap, ffi::OsStr, path::PathBuf, time::Duration};
+
+use crate::{
+    result::RunMeasurement,
+    time_bench::{Time, REPO_ROOT_PATH},
+};
+
+use tabled::settings::{
+    object::{Columns, Object, Rows},
+    width::MinWidth,
+    Alignment, Color, Modify, Style, Width,
+};
+use walkdir::WalkDir;
+
+/// A benchmark configuration and its resulting measurement.
+#[derive(Debug, Clone)]
+pub struct BenchmarkMeasurement {
+    command: Vec<String>,
+    params: String,
+    file: PathBuf,
+    measurement: RunMeasurement,
+}
+
+impl BenchmarkMeasurement {
+    /// Run flyvy on a single benchmark.
+    pub fn run(
+        command: Vec<String>,
+        args: Vec<String>,
+        file: PathBuf,
+        time_limit: Duration,
+    ) -> Self {
```

Comment:
Can you document the parameters to the `run` method? This will be useful for future developers who have to implement configurations for other subcommands of flyvy.

For example, my first question is: why is `command` a `Vec` and what's the difference between `command` and `args`?

Reply:
I added a complete docstring.

The split between `command`, `args`, and `file` was just to make presenting the results cleaner; they're actually just concatenated to form the command line. I could parse these out and just have the caller provide an array but (a) in practice I expect `command` is static, `args` is possible customized, and `file` is coming from a loop over all the examples so separate arguments makes sense and (b) I don't want to embed any logic for re-parsing our command line arguments.